Shape Up #1 - Fifth prototype fixes

lorenzo-cavazzi opened the pull request:
This PR includes the following:

- automatically select the first input or mapping on composite workflow (when available)
	* any workflow is ok for testing
- visualize links on composite workflows
	* example: https://renku-ci-ui-2093.dev.renku.ch/projects/workflows-pet-projects/workflows-parameters/workflows/952c938055a041168f6e795ae33b0d22
- improve auto-scroll behavior when workflow details become available
	* any project with workflow is ok for testing -- compare with an older version like https://renku-ci-ui-2038.dev.renku.ch
- (fix) visualize outdated workflows
	* example: https://renku-ci-ui-2093.dev.renku.ch/projects/workflows-pet-projects/nested-workflows/workflows/998ff6b7853d4889bbec55dfb31b4ba6


/deploy renku-core=workflow-ui-feature-branch #persist

ciyer commented on client/src/workflows/Workflows.present.tsx:
```diff
@@ -632,6 +607,45 @@ function VisualizerMappings({
+    return (<p className="m-2"><UnavailableDetail /></p>);
+
+  let num = 0;
+  let links = data.map((t: any) => {
```
If you provide a second argument, you can get the index in map.
```suggestion
  let links = data.map((t: any, num: number) => {
  
  }
```

lorenzo-cavazzi replied:
Ah, thanks! I added it in a separate commit since I will also initialize it

ciyer commented on client/src/workflows/Workflows.present.tsx:
```diff
@@ -632,6 +607,45 @@ function VisualizerMappings({
+  let links = data.map((t: any) => {
+    num++;
+    let targetCounter = 0;
+    const targets = t.sinks.map((target: any) => {
```
```suggestion
    const targets = t.sinks.map((target: any, targetCounter: number) => {
```

ciyer commented on client/src/workflows/Workflows.present.tsx:
```diff
@@ -760,6 +774,30 @@ function VisualizerDetailExpanded({ data, fullPath }: VisualizerDetailExpandedPr
+      workflows.list.find((w: any) => (w.id === data.plan_id)) :
+      null;
+    const subItem = data.id.replace(targetWorkflow.id + "/", "");
+    const newName = `[WF: ${targetWorkflow.name}] @ ${subItem.replace("/", " #")}`;
```
What does WF stand for? "Workflow"? I think it's ok to leave it as is ATM, but something that should be fixed before the feature is finalized.

lorenzo-cavazzi replied:
You are right, this is definitely not clear. We should come up with a better output
![image](https://user-images.githubusercontent.com/43481553/198386820-d6151d05-2354-4a53-9a77-e594bf7d725f.png)

ciyer reviewed (changes requested):
Overall, this looks good, and the things mentioned in the description work as explained.

The only thing I think should be changed in the code is that I think you can use `map` better to avoid having to introduce a counter. 

A minor comment: I find the "Loading" page a bit distracting

<img width="1335" alt="image" src="https://user-images.githubusercontent.com/1196411/198349942-bd8dbc62-fb6d-4156-9c2e-f7187f79ed13.png">

I think it could be improved by just removing the "Loading details" header and leaving just the spinner.

lorenzo-cavazzi commented:
Thanks for the review! I addressed the comments and removed the header above the spinner: that wasn't looking good. Here is the new output

![image](https://user-images.githubusercontent.com/43481553/198387787-79bc2e1f-838f-4b3b-88db-495ac4abbdbd.png)

ciyer commented on client/src/workflows/Workflows.present.tsx:
```diff
@@ -619,11 +619,9 @@ function VisualizerLinks({
-  let num = 0;
-  let links = data.map((t: any) => {
+  let links = data.map((t: any, num: number) => {
     num++;
```
This looks a bug!

ciyer reviewed (changes requested):
I like the improvements! I think one variable increment slipped through -- I think the `num++` should be removed now.
